### PR TITLE
[WIP] gstreamer1: update to 1.16.0.

### PR DESCRIPTION
**Pull request by Cogitri**
@pullmoll since I've removed the sed line for the pkgconfig file. The mentioned
include path does exist now, but maybe you could tell me which package you
had tested this with to make sure it's OK?

@xtraeme May I take maintainership of the gstreamer stack?

### CODE REVIEW

**Comment by xtraeme** [comment]
sure, go ahead

**Comment by Cogitri** [comment]
Thank you :)

**Comment by pullmoll** [comment]
@Cogitri are you sure you meant me? I don't recall having done anything in the gstreamer1 range.

**Comment by Cogitri** [comment]
I think 59bdd616c358ebaafc53dce2074aff4cba9de451 has been done by you?

**Comment by pullmoll** [comment]
Ohh, long time ago in a galaxy... yes, that was merely a hack around problems back then. Of course it should go away as soon as it can go away.

**Comment by Cogitri** [comment]
Alright thanks for confirming :)

**Comment by Cogitri** [comment]
@Kistelini the sndio patch doesn't build anymore. I've added the meson bits so you can test it (just enable the sndio option), but to be frank I don't feel like waiting much for a 1k LOC downstream patch. Please check if you want to work on this or I'll remove sndio support.

**Comment by Francesco149** [comment]
sndio patch fails to build because gstaudiosink.h is in ```gst-libs/gst/audio/```. according to the flags I'm seeing in the output it's not being added to the include dirs even though the meson.build file does have it

**Comment by Francesco149** [comment]
oh. actually, ```libsinc``` is never being passed to anything in the meson.build, so it could probably be fixed by patching meson.build or changing the patch to use a relative ```#include```

**Comment by Francesco149** [comment]
try this

```diff
diff --git a/srcpkgs/gst-plugins-base1/patches/sndio.patch b/srcpkgs/gst-plugins-base1/patches/sndio.patch
index 4aa7fdd3a0..f6dbb3cc92 100644
--- a/srcpkgs/gst-plugins-base1/patches/sndio.patch
+++ b/srcpkgs/gst-plugins-base1/patches/sndio.patch
@@ -524,7 +524,7 @@ index 0000000..3f9a9fe
 +  gstsndio = library('gstsndio',
 +    sndio_sources,
 +	c_args : gst_plugins_base_args,
-+	include_directories: configinc,
++	include_directories: [configinc, libsinc],
 +	dependencies: [gst_base_dep, sndio_dep],
 +    install : true,
 +    install_dir : plugins_install_dir,

```

**Comment by Francesco149** [comment]
ok this completely fixes the sndio patch

```diff
diff --git a/srcpkgs/gst-plugins-base1/patches/sndio.patch b/srcpkgs/gst-plugins-base1/patches/sndio.patch
index 4aa7fdd3a0..e32083d7f7 100644
--- a/srcpkgs/gst-plugins-base1/patches/sndio.patch
+++ b/srcpkgs/gst-plugins-base1/patches/sndio.patch
@@ -524,8 +524,8 @@ index 0000000..3f9a9fe
 +  gstsndio = library('gstsndio',
 +    sndio_sources,
 +	c_args : gst_plugins_base_args,
-+	include_directories: configinc,
-+	dependencies: [gst_base_dep, sndio_dep],
++	include_directories: [configinc, libsinc],
++	dependencies: [gst_base_dep, sndio_dep, audio_dep],
 +    install : true,
 +    install_dir : plugins_install_dir,
 +  )
```

**Comment by Cogitri** [comment]
Works, thanks @Francesco149. Guess I'll have to look more into the weirdness that C error messages are :)

File: srcpkgs/gst-plugins-base1/template [hunk 1]
```diff
@@ -1,59 +1,49 @@
...
 build_helper="gir"
-configure_args="--enable-experimental --disable-static
- $(vopt_enable gir introspection) $(vopt_enable cdparanoia) $(vopt_enable sndio)"
-hostmakedepends="automake gettext-devel libtool python pkg-config
- glib-devel"
+configure_args="-Dtremor=disabled -Dexamples=disabled -Dgtk_doc=disabled
+ -Dcdparanoia=$(vopt_if cdparanoia enabled disabled)
+ -Dintrospection=$(vopt_if gir enabled disabled)
+ -Dsndio=$(vopt_if sndio enabled disabled)"
+hostmakedepends="gettext-devel pkg-config glib-devel"
```
Comment (maxice8) [inline srcpkgs/gst-plugins-base1/template]: needs `orcc` on hostmakedepends=

**Comment by jnbr** [comment]
There are a few more related packages which should be bumped as well.
Easiest way to find them is to grep for the version:
```
grep -r "^version=1.14.4$" srcpkgs
srcpkgs/gst-libav/template:version=1.14.4
srcpkgs/gst-omx/template:version=1.14.4
srcpkgs/gst-plugins-bad1/template:version=1.14.4
srcpkgs/gst-plugins-base1/template:version=1.14.4
srcpkgs/gst-plugins-good1/template:version=1.14.4
srcpkgs/gst-plugins-ugly1/template:version=1.14.4
srcpkgs/gst1-editing-services/template:version=1.14.4
srcpkgs/gst1-python/template:version=1.14.4
srcpkgs/gst1-python3/template:version=1.14.4
srcpkgs/gstreamer-vaapi/template:version=1.14.4
srcpkgs/gstreamer1/template:version=1.14.4
```

**Comment by jnbr** [comment]
What is the status on this?

**Comment by maxice8** [comment]
@jnbr carry it on, it won't be recieving new stuff as far as i am aware.

**Comment by Cogitri** [comment]
Yup, feel free to use the commits if they're useful though :)

**Pull request closed without merging**